internal/controller/warehouses/git.go

```diff
@@ -389,6 +414,37 @@ func matchesRegexpList(stringToMatch string, regexpList []*regexp.Regexp) bool {
 	return foundMatch
 }
 
+// matchesGlobList is a general purpose function iterating given slice of
+// strings (globList) to check if any of globs match stringToMatch string,
+// if match is found it returns true and nil, if match is not found it
+// returns false and nil and if it finds a malformed glob pattern - false
+// and ErrBadPattern is returned.
+func matchesGlobList(stringToMatch string, globList []string) (bool, error) {
+	for _, glob := range globList {
+		match, err := filepath.Match(glob, stringToMatch)
+		if err != nil {
+			return false, errors.New("syntax error in pattern: " + glob)
+		}
+		if match {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// matchesPrefixList is a general purpose function iterating given slice of
+// strings (prefixList) to check if any of prefixes match stringToMatch string,
+// if match is found it returns true, if match is not found it
+// returns false
+func matchesPrefixList(stringToMatch string, prefixList []string) bool {
+	for _, prefix := range prefixList {
+		if strings.HasPrefix(stringToMatch, prefix) {
```

Comment:
This doesn't look right to me...

This would have `foobar` matching `foo` and I don't think that's what we really intended. Perhaps "prefix" is a poor word choice? Because all along, I thought we've been talking about something more like this:

```golang
func matchesPrefix(path string, prefixes []string) (bool, error) {
	for _, prefix := range prefixes {
		relPath, err := filepath.Rel(prefix, path)
		if err != nil {
			return false, err
		}
		if !strings.Contains(relPath, "..") {
			return true, nil
		}
	}
	return false, nil
}
```